In [1]:
import numpy as np
import pandas as pd
import math
import matplotlib.pyplot as plt

In [7]:
data09 = pd.read_excel('DDS9_Data_Extract_with_labels.xlsx', encoding='latin1')
data10 = pd.read_excel('DDS10_Data_Extract_with_labels.xlsx', encoding='latin1')
data11 = pd.read_excel('DDS11_Data_Extract_with_labels.xlsx', encoding='latin1')

In [8]:
print(data09.shape)
print(data10.shape)
print(data11.shape)

(2076, 191)
(2205, 197)
(2131, 198)


By printing the shape of the three year's worth of data we can see two important pieces of take-aways:
1. Each of the years has a similar number of observations. This is good as it means we won't be overreliant on one year's data which may have undue influence on the results.
2. We have a different number of features for each year. Having 1/3 or more of observations automatically not having any data for one of the features is a problem.

In order to account for 2, the first step will be to identify which features are not shared between the years.

In [38]:
cols9, cols10, cols11 = list(data09.columns), list(data10.columns), list(data11.columns)
cols910 = [x for x in cols9 if x in cols10]
shared_cols = [x for x in cols910 if x in cols11]
len(shared_cols)

109

Here we can see that there are 109 shared columns between the three. Next is to investigate the columns that are not shared and see what there is to learn about the columns not in the list.

In [36]:
not_cols910 = [x for x in cols9 if x not in cols10]
not_cols109 = [x for x in cols10 if x not in cols9]
not_cols_comb910 = not_cols910 + not_cols109

not_cols_91011 = [x for x in not_cols_comb910 if x not in cols11]
not_cols_11910 = [x for x in cols11 if x not in not_cols_comb910]

not_cols = not_cols_91011 + not_cols_11910
len(not_cols)

204

In [40]:
##### LIST OF SHARED COLUMNS ######

sorted(shared_cols)

['Q10 - Of those products you indicated you do not currently own, which of the following do you plan to purchase in the next 12 months?-Blu-ray disc player/DVD player',
 'Q10 - Of those products you indicated you do not currently own, which of the following do you plan to purchase in the next 12 months?-Computer network/router in your home for wireless computer/laptop usage',
 'Q10 - Of those products you indicated you do not currently own, which of the following do you plan to purchase in the next 12 months?-Dedicated e-book reader',
 'Q10 - Of those products you indicated you do not currently own, which of the following do you plan to purchase in the next 12 months?-Desktop computer',
 'Q10 - Of those products you indicated you do not currently own, which of the following do you plan to purchase in the next 12 months?-Digital video recorder (DVR)',
 "Q10 - Of those products you indicated you do not currently own, which of the following do you plan to purchase in the next 12 months?-D

In [37]:
#### NOT SHARED COLUMNS ####

sorted(not_cols)

['FINAL WEIGHTS',
 'Final Weighting',
 'Q10 - Of those products you indicated you do not currently own, which of the following do you plan to purchase in the next 12 months?-3D printer',
 'Q10 - Of those products you indicated you do not currently own, which of the following do you plan to purchase in the next 12 months?-Basic mobile phone',
 'Q10 - Of those products you indicated you do not currently own, which of the following do you plan to purchase in the next 12 months?-Blu-ray disc player/DVD player',
 'Q10 - Of those products you indicated you do not currently own, which of the following do you plan to purchase in the next 12 months?-Computer network/router in your home for wireless computer/laptop usage',
 'Q10 - Of those products you indicated you do not currently own, which of the following do you plan to purchase in the next 12 months?-Dedicated e-book reader',
 'Q10 - Of those products you indicated you do not currently own, which of the following do you plan to purchase in